src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -1055,22 +1055,31 @@ void Replayer<I>::remove_mirror_peer_uuid(const std::string &snap_id) {
       });
 
   if (remote_snap == m_remote_group_snaps.end()) {
+    derr << "snapshot not found: snap_id: "
+         << snap_id << dendl;
     return;
   }
 
   auto rns = std::get_if<cls::rbd::GroupSnapshotNamespaceMirror>(
       &remote_snap->snapshot_namespace);
-  if (rns != nullptr) {
-    rns->mirror_peer_uuids.clear();
+  if (rns == nullptr) {
+    derr << "not a mirror group snapshot: snap_id: "
```

Comment:
Nit: Doesn't look like a helpful msg, hence this condition can be merged with the next one as I mentioned before, but I will leave it to you if you still want to go with it.

Reply:
It's always better to have a 1:1 mapping between the code path/error condition and the error message so that one doesn't have to guess which check was hit upon getting a log snippet.  Something like "remote group snapshot is not a mirror snapshot" for this one and "remote group snapshot not found" for the other one might make it a bit better though.